### PR TITLE
[ISSUE #6136]Add test case for GetMaxOffsetRequestHeader

### LINKED ISSUE #6136: [Test🧪] Add test case for GetMaxOffsetRequestHeader
### Prerequisites

- [x] I have searched existing test issues

### Test Scope

Add test case for GetMaxOffsetRequestHeader

### Test Cases

Add test case for GetMaxOffsetRequestHeader

### Willing to contribute?

- [ ] I can submit a PR for this

### PR DESCRIPTION
**Pull request by Andra-Fauzi**
## Target Branch

- Fix #6136

## Which Issue(s) This PR Fixes(Closes)
- Relates to comprehensive test coverage for GetMaxOffsetRequestHeader

## Brief Description
Added comprehensive test coverage for `GetMaxOffsetRequestHeader` struct including:
- Tests for required fields only
- Tests for all fields with nested topic request headers
- Edge case tests (empty topic, long string values)
- Tests for all trait implementation methods and setters

## How Did You Test This Change?
Ran all unit tests:
```bash
cargo test -p rocketmq-remoting

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit tests covering construction, required and optional fields, nested header interactions, accessors/mutators, empty and long values, and boundary/error cases for the GetMaxOffsetRequestHeader to improve reliability and coverage.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: rocketmq-remoting/tests/test_get_max_offset_request_header.rs
Comment (WaterWhisperer) [inline rocketmq-remoting/tests/test_get_max_offset_request_header.rs:1]: the issue means adding unit tests to the get_max_offset_request_header.rs file, just like any other file under the module

File: rocketmq-remoting/tests/test_get_max_offset_request_header.rs [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+            topic_request_header: None,
+        };
+        assert_eq!(header.topic, CheetahString::from(&long_string));
+        assert_eq!(header.queue_id, 1);
+        assert!(header.committed);
+        assert!(header.topic_request_header.is_none());
+    }
+
+    #[test]
+    fn test_fn_lo() {
```
Comment (WaterWhisperer) [inline rocketmq-remoting/tests/test_get_max_offset_request_header.rs]: IMO it's best to omit the `test_` prefix from the test function names, since these functions are already in the `tests` module.

File: rocketmq-remoting/tests/test_get_max_offset_request_header.rs [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+            queue_id: 1,
+            committed: true,
+            topic_request_header: Some(topic_req),
+        };
+
+        assert_eq!(header.lo(), Some(true));
+    }
+
+    #[test]
+    fn test_fn_set_lo() {
```
Comment (WaterWhisperer) [inline rocketmq-remoting/tests/test_get_max_offset_request_header.rs]: ditto

Comment (Andra-Fauzi) [reply rocketmq-remoting/tests/test_get_max_offset_request_header.rs]: Thanks for your review.
